### PR TITLE
Update DeveloperGuide.md for Update product

### PR DESCRIPTION
**Pull request by qlchan24**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -159,6 +159,46 @@ After the `edit` command is called, the user input will be sent to **EditCommand
...
+After which, it will call the `UpdateCommand` with the `Type`, warehouse/supplier's `Name` and `Product` created, 
+and return it to `ClinicParser#Parse` which will in turn return the `UpdateCommand` to `LogicManager#execute`.
+
+The following sequence diagram shows how the update product operation works: (TODO: Insert diagram)
+
+Step 3. `LogicManager#execute` calls `UpdateCommand#execute` with the `Model` instance. In `LogicManager#execute`, 
+the `Model#getWarehouseByName` or `Model#getSupplierByName` is called (to be implemented), which 
+iterates through the warehouse/supplier list to find a warehouse/supplier with a `Name` that matches the one provided 
+in the `UpdateCommand`. If it is not found, `NoSuchElementException` is thrown, otherwise, the `UpdateCommand#execute` 
+method creates a new `Set<Product>` based on the existing product set for that warehouse/supplier.
```
Comment (Criss-Wang) [inline docs/DeveloperGuide.md]: Do you mean the `Set<Product>` is identical to the existing product set here?

**Review by Criss-Wang — COMMENTED** [review]
Nice description!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -159,6 +159,46 @@ After the `edit` command is called, the user input will be sent to **EditCommand
...
+
+Given below is an example usage scenario and how the update product mechanism behaves at each step.
+
+Step 1. The user decides to update the stock for a product called 'Panadol' with a new quantity of 50 units 
+in the warehouse named 'Jurong Warehouse' by executing the `update ct/w n/Jurong Warehouse pd/Panadol q/50` command.
+The `ClinicParser#parseCommand` will then call the `UpdateCommandParser#parse` method with all the arguments 
+passed by the user.
+ 
+Step 2. `UpdateCommandParser#parse` then attempts to create new `Name` instances for the supplier/warehouse 
+and the product, and new `tag` instances for the product. An exception will be thrown if any of the names or
```
Comment (zhengweii) [inline docs/DeveloperGuide.md:223]: Hmm the example above doesn't really show the t/TAG! Might be a little confusing!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (qlchan24) [reply docs/DeveloperGuide.md]: Yes, thats what I meant.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (qlchan24) [reply docs/DeveloperGuide.md:223]: Thats true, Ill update the example accordingly. Thanks!

**Review by jeffreytjs — APPROVED** [review]
LGTM

**Pull request merged**